File: app/src/main/java/com/example/helpwheel/ui/notes/notesFragments/AddNotesFragment.java
package com.example.helpwheel.ui.notes.notesFragments;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.fragment.NavHostFragment;

import com.example.helpwheel.R;
import com.example.helpwheel.databinding.FragmentAddNotesBinding;
import com.example.helpwheel.ui.notes.databases.DatabaseClass;
import com.google.android.material.datepicker.MaterialDatePicker;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.TimeZone;


public class AddNotesFragment extends Fragment {
    EditText title, description, webURL;
    Button addNote;
    FragmentAddNotesBinding binding;
    private MaterialDatePicker<Long> materialDatePicker;
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = FragmentAddNotesBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        MaterialDatePicker.Builder<Long> materialDateBuilder = MaterialDatePicker.Builder.datePicker();
        materialDateBuilder.setTitleText("SELECT A DATE");
        materialDatePicker = materialDateBuilder.build();
        title = binding.title;
        description = binding.description;
        addNote = binding.addNote;
        webURL = binding.webUlr;
        addNote.setOnClickListener(v -> {
            if (!TextUtils.isEmpty(title.getText().toString()) && !TextUtils.isEmpty(description.getText().toString())) {
                String descUrl = description.getText().toString().trim() + " " + webURL.getText().toString().trim();
                DatabaseClass db = new DatabaseClass(getContext());
                db.addNotes(title.getText().toString().trim(), descUrl);
                NavHostFragment.findNavController(this).navigate(R.id.action_addNotesFragment_to_navigation_home);
            }
            if (title.getText().toString().isEmpty()){
                binding.titleEditText.setError(getString(R.string.field_must_be_filled));
            }
            if (description.getText().toString().isEmpty()){
                binding.descriptionEditText.setError(getString(R.string.field_must_be_filled));
            }

        });
        binding.descriptionEditText.setEndIconOnClickListener(v1 -> materialDatePicker.show(getChildFragmentManager(), "MATERIAL_DATE_PICKER"));

        materialDatePicker.addOnPositiveButtonClickListener(this::getDate);
    }
    private void getDate(Long selection){
        Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
        calendar.setTimeInMillis(selection);
        SimpleDateFormat format = new SimpleDateFormat("dd.MM.yyyy");
        String formattedDate  = format.format(calendar.getTime());
        binding.description.setText(formattedDate);
    }

    @Override
    public void onStart() {
        super.onStart();
        requireActivity().findViewById(R.id.customBnb).setVisibility(View.GONE);
    }
}

File: app/src/main/java/com/example/helpwheel/ui/notes/notesFragments/add/AddNotesFragment.kt
package com.example.helpwheel.ui.notes.notesFragments.add
import android.annotation.SuppressLint
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.cardview.widget.CardView
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.navigation.fragment.NavHostFragment
import com.example.helpwheel.R
import com.example.helpwheel.databinding.FragmentAddNotesBinding
import com.example.helpwheel.ui.notes.model.NoteModel
import java.text.SimpleDateFormat
import java.util.*


class AddNotesFragment: Fragment() {
    private lateinit var binding: FragmentAddNotesBinding
    private lateinit var addNotesViewModel: AddNotesViewModel
   // private lateinit var materialDatePicker: MaterialDatePicker<Long>

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentAddNotesBinding.inflate(inflater, container, false)
        addNotesViewModel = ViewModelProvider(this)[AddNotesViewModel::class.java]
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        //materialDatePicker = MaterialDatePicker()
        initListeners()
    }

    override fun onStart() {
        super.onStart()
        requireActivity().findViewById<CardView>(R.id.customBnb).visibility = View.GONE
    }

    @SuppressLint("SimpleDateFormat")
    private fun getDate(selection: Long) {
        val calendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
        calendar.timeInMillis = selection
        val format = SimpleDateFormat("dd.MM.yyyy")
        val formattedDate = format.format(calendar.time)
        binding.description.setText(formattedDate)
    }



    private fun initListeners() {
        binding.addNote.setOnClickListener{
            val title = binding.title
            val description = binding.description
            val webURL = binding.webUlr
            if (title.text.toString().isNotEmpty() && description.text.toString().isNotEmpty() && webURL.text.toString().isNotEmpty()) {
                addNotesViewModel.insert(NoteModel(title = title.text.toString(), description = description.text.toString(), web_site = webURL.toString())) {}
                NavHostFragment.findNavController(this).navigate(R.id.action_addNotesFragment_to_navigation_home)
            } else if (title.text.toString().isNotEmpty() && description.text.toString().isNotEmpty()) {
                addNotesViewModel.insert(NoteModel(title = title.text.toString(), description = description.text.toString())) {}
                NavHostFragment.findNavController(this).navigate(R.id.action_addNotesFragment_to_navigation_home)
            }
            if (title.text.toString().isEmpty())
                binding.titleEditText.error = getString(R.string.field_must_be_filled)
            if (description.text.toString().isEmpty())
                binding.descriptionEditText.error = getString(R.string.field_must_be_filled)
        }
        //binding.descriptionEditText.setEndIconOnClickListener{materialDatePicker.show(childFragmentManager, "MATERIAL_DATE_PICKER")}
        binding.description.setOnClickListener{ binding.descriptionEditText.error = null }
        //materialDatePicker.addOnPositiveButtonClickListener(this::getDate)
    }
}
